packages/f/ffmpeg/xmake.lua

```diff
@@ -45,6 +45,33 @@ package("ffmpeg")
         add_deps("yasm")
     end
 
+    if on_fetch then
+        on_fetch("mingw", "linux", "macosx", function (package, opt)
+            import("lib.detect.find_tool")
+            if opt.system then
+                local result
+                for _, name in ipairs({"libavcodec", "libavdevice", "libavfilter", "libavformat", "libavutil", "libpostproc", "libswresample", "libswscale"}) do
+                    local pkginfo = package:find_package("pkgconfig::" .. name, opt)
+                    if pkginfo then
+                        pkginfo.version = nil
+                        if not result then
+                            result = pkginfo
+                        else
+                            result = result .. pkginfo
+                        end
+                    else
+                        return {}
+                    end
+                end
+                local ffmpeg = find_tool("ffmpeg", {check = "-help", version = true, command = "-version", parse = "%d+%.?%d+%.?%d+", force = true})
```

Comment:
```lua
if result then
    local ffmpeg = ...
end

Reply:
> Because of the else inside the loop, result is always defined when the code here is executed, so no need of the if here right ?

if all sub-libraries are not found, pkginfo is always nil, so result is nil too.

Reply:
As soon as one pkginfo is nil, the return at line 63 will be triggered and we will never arrives here.